source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -372,10 +370,15 @@ namespace PlayFab
     void PlayFabWinHttpPlugin::SetPredefinedHeaders(const CallRequestContainer& requestContainer, HINTERNET hRequest)
     {
         UNREFERENCED_PARAMETER(requestContainer);
-        WinHttpAddRequestHeaders(hRequest, L"Accept: application/json", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"Content-Type: application/json; charset=utf-8", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str(), -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"X-ReportErrorAsSuccess: true", -1, 0);
+        headerNoErrors = !TryAddHeader(hRequest, L"Accept: application/json", -1, 0) &&
+            !TryAddHeader(hRequest, L"Content-Type: application/json; charset=utf-8", -1, 0) &&
+            !TryAddHeader(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str(), -1, 0) &&
+            !TryAddHeader(hRequest, L"X-ReportErrorAsSuccess: true", -1, 0);
+    }
+
+    bool PlayFabWinHttpPlugin::TryAddHeader(HINTERNET hRequest, LPCWSTR lpszHeaders, DWORD dwHeadersLength, DWORD dwModifiers)
+    {
+        return WinHttpAddRequestHeaders(hRequest, lpszHeaders, dwHeadersLength, dwModifiers) == S_OK;
```

Comment:
WinHttpAddRequestHeaders does not return an HRESULT. [It returns a bool](https://docs.microsoft.com/en-us/windows/win32/api/winhttp/nf-winhttp-winhttpaddrequestheaders).

Reply:
which means any time we successfully add headers this will return false (since S_OK is 0). which means I think this breaks even the golden path success case